platform/windows/os_windows.cpp

```diff
@@ -914,6 +916,14 @@ Error OS_Windows::set_cwd(const String &p_cwd) {
 	return OK;
 }
 
+String OS_Windows::get_cwd()
+{
+	WCHAR ret[2048];
+	GetCurrentDirectoryW(2048, ret);
+	String cwd = String::utf16((const char16_t *)ret);
+	return cwd;
```

Comment:
I think we are using 2048 buffer for the same function, but some places of the Windows API mention 32K as max. path length. I'm not sure if it's relevant in this case, but it might be more robust to use "preflighting" syntax and allocate specific buffer size:

```c++
	const DWORD size = GetCurrentDirectoryW(0, nullptr);

	Char16String buffer;
	buffer.resize((int)size);
	if (GetCurrentDirectoryW(size, (wchar_t *)buffer.ptrw()) == 0) {
		return OS::get_cwd();
	}

	String result;
	result.parse_utf16(buffer.ptr());

	return result;
```